### PR TITLE
small refactor of bucket configuration

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:** refactor to add convenience function.

Originally I wrote this to be used in the distribution API (but will probably use a different method).  This refactor cleans up some repeated code.

Addresses [CUMULUS-1223: public s3 urls](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1223)

## Changes

* Adds new function `@cumulus/common/bucketsConfigJsonObject` which reads the buckets.json file from the stack configuration's system_bucket and returns an object suitable for input to `BucketsConfig`.
* Adds helpers to BucketsConfig to get lists of buckets by types.

## PR Checklist

- [ X ] Update CHANGELOG
- [ X ] Unit tests
- [ X ] Integration tests

### CODE REVIEW

File: packages/common/bucketsConfigJsonObject.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const aws = require('./aws');
+/**
```
Comment (markdboyd) [inline packages/common/bucketsConfigJsonObject.js]: Super nitpick: add a space before the JSdoc

File: packages/common/bucketsConfigJsonObject.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * @param {string} bucket - system bucket name.
+ * @param {string} stackName - stack name.
+ * @returns {Object} - stack's bucket configuration.
+ */
+async function bucketsConfigJsonObject(
+  bucket = process.env.system_bucket,
+  stackName = process.env.stackName
+) {
+  const bucketsString = await aws.s3().getObject({
```
Comment (markdboyd) [inline packages/common/bucketsConfigJsonObject.js]: I was going to ask if we should validate the `bucket` and `stackName` arguments, but if they're empty then S3 will probably return a helpful enough error message to make that clear?

File: packages/common/bucketsConfigJsonObject.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+const aws = require('./aws');
+/**
+ * Retrieve the stack's bucket configuration from s3 and return the bucket configuration object.
+ *
+ * @param {string} bucket - system bucket name.
+ * @param {string} stackName - stack name.
+ * @returns {Object} - stack's bucket configuration.
+ */
+async function bucketsConfigJsonObject(
```
Comment (markdboyd) [inline packages/common/bucketsConfigJsonObject.js]: Should we unit test this by staging a JSON file in localstack?

File: packages/common/bucketsConfigJsonObject.js (hunk 2 above)
Comment (flamingbear) [reply packages/common/bucketsConfigJsonObject.js]: Yes, but I added a bit of code to make it even clearer.

File: packages/common/bucketsConfigJsonObject.js (hunk 3 above)
Comment (flamingbear) [reply packages/common/bucketsConfigJsonObject.js]: It seems a little testing AWS, but sure.

Comment (flamingbear) [reply packages/common/bucketsConfigJsonObject.js]: c4c5aed43

File: packages/common/bucketsConfigJsonObject.js (hunk 2 above)
Comment (flamingbear) [reply packages/common/bucketsConfigJsonObject.js]: c4c5aed43

File: packages/common/tests/test-bucketsConfigJsonObject.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  t.deepEqual(bucketConfig, actualBucketObject);
+});
+
+test('has understandable error messages for bad bucket name', async (t) => {
+  process.env.system_bucket = 'badbucket';
+  process.env.stackName = context.stackName;
+
+  await t.throws(
+    bucketsConfigJsonObject(),
+    'Unable to read bucketsConfiguration from s3: The specified bucket does not exist'
```
Comment (markdboyd) [inline packages/common/tests/test-bucketsConfigJsonObject.js]: Is there any way to include the actual bucket name in the error message?

Comment (flamingbear) [reply packages/common/tests/test-bucketsConfigJsonObject.js]: sure. 2bbd78eab

**Pull request merged**